test/clj/rems/api/test_applications.clj

```diff
@@ -489,6 +489,12 @@
                (authenticate api-key "bob")
                handler))))
 
+    (testing "seeing draft application as reporter is forbidden"
+      (is (response-is-forbidden?
+            (-> (request :get (str "/api/applications/" application-id))
+                (authenticate api-key "reporter")
+                handler))))
+
```

Comment:
Could combine the two seeing is forbidden? Just reduce the duplication a bit or introduce even more cases?